motion_primitives_forward_controller/test/test_motion_primitives_forward_controller.cpp

```diff
@@ -0,0 +1,277 @@
+//   static constexpr double TEST_DISPLACEMENT = 23.24;
+//   std::shared_ptr<ControllerReferenceMsg> msg = std::make_shared<ControllerReferenceMsg>();
+//   // When slow mode is enabled command ends up being half of the value
+//   msg->joint_names = joint_names_;
+//   msg->displacements.resize(joint_names_.size(), TEST_DISPLACEMENT);
+//   msg->velocities.resize(joint_names_.size(), std::numeric_limits<double>::quiet_NaN());
+//   msg->duration = std::numeric_limits<double>::quiet_NaN();
+//   controller_->input_ref_.writeFromNonRT(msg);
+//   controller_->control_mode_.writeFromNonRT(control_mode_type::SLOW);
+
+//   EXPECT_EQ(*(controller_->control_mode_.readFromRT()), control_mode_type::SLOW);
+//   ASSERT_EQ((*(controller_->input_ref_.readFromRT()))->displacements[0], TEST_DISPLACEMENT);
+//   ASSERT_EQ(
+//     controller_->update(rclcpp::Time(0), rclcpp::Duration::from_seconds(0.01)),
+//     controller_interface::return_type::OK);
+
+//   EXPECT_EQ(joint_command_values_[STATE_MY_ITFS], TEST_DISPLACEMENT / 2);
+//   EXPECT_TRUE(std::isnan((*(controller_->input_ref_.readFromRT()))->displacements[0]));
+// }
+
+// TEST_F(MotionPrimitivesForwardControllerTest, publish_status_success)
+// {
+//   SetUpController();
+
+//   ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), NODE_SUCCESS);
+//   ASSERT_EQ(controller_->on_activate(rclcpp_lifecycle::State()), NODE_SUCCESS);
+
+//   ASSERT_EQ(
+//     controller_->update(rclcpp::Time(0), rclcpp::Duration::from_seconds(0.01)),
+//     controller_interface::return_type::OK);
+
+//   ControllerStateMsg msg;
+//   subscribe_and_get_messages(msg);
+
+//   ASSERT_EQ(msg.set_point, 101.101);
+// }
+
+// TEST_F(MotionPrimitivesForwardControllerTest, receive_message_and_publish_updated_status)
+// {
+//   SetUpController();
+//   rclcpp::executors::MultiThreadedExecutor executor;
+//   executor.add_node(controller_->get_node()->get_node_base_interface());
+
+//   ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), NODE_SUCCESS);
+//   ASSERT_EQ(controller_->on_activate(rclcpp_lifecycle::State()), NODE_SUCCESS);
+
+//   ASSERT_EQ(
+//     controller_->update(rclcpp::Time(0), rclcpp::Duration::from_seconds(0.01)),
+//     controller_interface::return_type::OK);
+
+//   ControllerStateMsg msg;
+//   subscribe_and_get_messages(msg);
+
+//   ASSERT_EQ(msg.set_point, 101.101);
+
+//   publish_commands();
+//   ASSERT_TRUE(controller_->wait_for_commands(executor));
+
+//   ASSERT_EQ(
+//     controller_->update(rclcpp::Time(0), rclcpp::Duration::from_seconds(0.01)),
+//     controller_interface::return_type::OK);
+
+//   EXPECT_EQ(joint_command_values_[CMD_MY_ITFS], 0.45);
+
+//   subscribe_and_get_messages(msg);
+
+//   ASSERT_EQ(msg.set_point, 0.45);
+// }
+
+// int main(int argc, char ** argv)
+// {
+//   ::testing::InitGoogleTest(&argc, argv);
+//   rclcpp::init(argc, argv);
+//   int result = RUN_ALL_TESTS();
+//   rclcpp::shutdown();
+//   return result;
+// }
```

Comment:
2 problems:
* it's all commented out :D 
* it's always better to use gmock, not gtest. You get much better prints.